addons/account/wizard/account_resequence.py

```diff
@@ -151,6 +151,8 @@ def resequence(self):
             if self.ordering == 'date':
                 raise UserError(_('You can not reorder sequence by date when the journal is locked with a hash.'))
         moves_to_rename = self.env['account.move'].browse(int(k) for k in new_values.keys())
+        moves_to_reset_payment_reference = moves_to_rename.filtered(lambda l: l.payment_reference == l._get_invoice_computed_reference())
+        moves_to_reset_payment_reference.payment_reference = False
```

Comment:
I put the payment_reference to False so the compute is triggered and changes the reference...
However it may be a good idea to put the change in write..
@smetl  what do you think ?